velox/dwio/dwrf/common/FileMetadata.h

```diff
@@ -940,6 +1003,1005 @@ class StripeFooterWrapper : public ProtoWrapperBase {
+  }
+
+ private:
+  // private helper with no format checking
+  inline proto::DoubleStatistics* dwrfPtr() const {
+    return reinterpret_cast<proto::DoubleStatistics*>(rawProtoPtr());
+  }
+  inline proto::orc::DoubleStatistics* orcPtr() const {
+    return reinterpret_cast<proto::orc::DoubleStatistics*>(rawProtoPtr());
+  }
+};
+
+class StringStatisticsWriteWrapper : public ProtoWriteWrapperBase {
+ public:
+  explicit StringStatisticsWriteWrapper(
+      proto::StringStatistics* stringStatistics)
+      : ProtoWriteWrapperBase(DwrfFormat::kDwrf, stringStatistics) {}
+
+  explicit StringStatisticsWriteWrapper(
+      proto::orc::StringStatistics* stringStatistics)
+      : ProtoWriteWrapperBase(DwrfFormat::kOrc, stringStatistics) {}
+
+  void setSum(uint64_t sum) {
+    format_ == DwrfFormat::kDwrf ? dwrfPtr()->set_sum(sum)
+                                 : orcPtr()->set_sum(sum);
+  }
+
+  void setMinimum(std::string minimum) {
+    format_ == DwrfFormat::kDwrf ? dwrfPtr()->set_minimum(minimum)
+                                 : orcPtr()->set_minimum(minimum);
+  }
+
+  void setMaximum(std::string maximum) {
+    format_ == DwrfFormat::kDwrf ? dwrfPtr()->set_maximum(maximum)
+                                 : orcPtr()->set_maximum(maximum);
+  }
+
+ private:
+  // private helper with no format checking
+  inline proto::StringStatistics* dwrfPtr() const {
+    return reinterpret_cast<proto::StringStatistics*>(rawProtoPtr());
+  }
+  inline proto::orc::StringStatistics* orcPtr() const {
+    return reinterpret_cast<proto::orc::StringStatistics*>(rawProtoPtr());
+  }
+};
+
+class BinaryStatisticsWriteWrapper : public ProtoWriteWrapperBase {
+ public:
+  explicit BinaryStatisticsWriteWrapper(
+      proto::BinaryStatistics* binaryStatistics)
+      : ProtoWriteWrapperBase(DwrfFormat::kDwrf, binaryStatistics) {}
+
+  explicit BinaryStatisticsWriteWrapper(
+      proto::orc::BinaryStatistics* binaryStatistics)
+      : ProtoWriteWrapperBase(DwrfFormat::kOrc, binaryStatistics) {}
+
+  void setSum(uint64_t sum) {
+    format_ == DwrfFormat::kDwrf ? dwrfPtr()->set_sum(sum)
+                                 : orcPtr()->set_sum(sum);
+  }
+
+ private:
+  // private helper with no format checking
+  inline proto::BinaryStatistics* dwrfPtr() const {
+    return reinterpret_cast<proto::BinaryStatistics*>(rawProtoPtr());
+  }
+  inline proto::orc::BinaryStatistics* orcPtr() const {
+    return reinterpret_cast<proto::orc::BinaryStatistics*>(rawProtoPtr());
+  }
+};
+
+class ColumnStatisticsWriteWrapper : public ProtoWriteWrapperBase {
+ public:
+  explicit ColumnStatisticsWriteWrapper(proto::ColumnStatistics* footer)
+      : ProtoWriteWrapperBase(DwrfFormat::kDwrf, footer) {}
+
+  explicit ColumnStatisticsWriteWrapper(proto::orc::ColumnStatistics* footer)
+      : ProtoWriteWrapperBase(DwrfFormat::kOrc, footer) {}
+
+  void reset(const proto::ColumnStatistics* dwrfStatistics) {
+    VELOX_CHECK_EQ(format_, DwrfFormat::kDwrf);
```

Comment:
@wypb, prob good to do a null check here as well! And same thing for stats write wrapper.